R/mobr.R

```diff
@@ -286,12 +291,10 @@ rescale_rarefaction = function(comm, ratio, effort){
 deltaS_N = function(comm, ref_dens, inds){
     nplots = nrow(comm)
     group_dens = sum(comm) / nplots
-    ratio = ref_dens / group_dens
-    interp_S_samp = rescale_rarefaction(comm, ratio, inds)
-    # Ensure that interp_S_samp has the right length (filled with NA's if needed)
-    interp_S_samp = interp_S_samp[1:length(inds)]
-    S_indiv = rarefaction(comm, 'indiv', inds)
-    deltaS = interp_S_samp - S_indiv
+    dens_ratio = ref_dens / group_dens
+    S_rescaled = rarefaction(comm, 'indiv', inds, dens_ratio=dens_ratio)
```

Comment:
I don't understand why `S_rescaled = S_rescaled[1:length(inds)]` is needed because `S_rescaled` and `S_raw` will be evaluated at `inds`. For example
```r
rarefaction(inv_comm, 'indiv', 1:10)
       1        2        3        4        5        6        7 
1.000000 1.678113 2.203415 2.652627 3.060896 3.444610 3.811668 
       8        9       10 
4.166095 4.510125 4.845139 
rarefaction(inv_comm, 'indiv', 1:10, dens_ratio=.25)
        1         2         3         4         5         6 
 2.652627  4.166095  5.491725  6.707672  7.842070  8.912143 
        7         8         9        10 
9.929240 10.901120 11.833309 12.729913 
rarefaction(inv_comm, 'indiv', 1:10, dens_ratio=1.4)
        1         2         3         4         5         6 
0.7582386 1.3171502 1.7600487 2.1342170 2.4665704 2.7725048 
        7         8         9        10 
3.0608960 3.3369237 3.6036672 3.8630078 
```
All those functions return values for 1:10. Am I missing something?